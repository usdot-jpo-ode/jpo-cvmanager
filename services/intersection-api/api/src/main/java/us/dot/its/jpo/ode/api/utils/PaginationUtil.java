package us.dot.its.jpo.ode.api.utils;

import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

public class PaginationUtil {

    public static <T> ResponseEntity<List<T>> createResponseEntityWithPaginationHeaders(Page<T> page) {
        HttpHeaders headers = new HttpHeaders();
        headers.add("X-Total-Count", String.valueOf(page.getTotalElements()));
        headers.add("X-Total-Pages", String.valueOf(page.getTotalPages()));
        headers.add("X-Has-More", String.valueOf(page.hasNext()));
        return new ResponseEntity<List<T>>(page.getContent(), headers, HttpStatus.OK);
    }
}