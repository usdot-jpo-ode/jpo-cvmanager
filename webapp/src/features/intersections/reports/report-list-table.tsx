import React from 'react'
import { format } from 'date-fns'
import {
  Avatar,
  Box,
  Button,
  IconButton,
  Table,
  TableBody,
  TableCell,
  TablePagination,
  TableRow,
  Typography,
} from '@mui/material'
import PerfectScrollbar from 'react-perfect-scrollbar'
import { ArrowDownward } from '@mui/icons-material'
import toast from 'react-hot-toast'
import ReportsApi, { ReportMetadata } from '../../../apis/intersections/reports-api'
import { useSelector } from 'react-redux'
import { useNavigate } from 'react-router-dom'
import { selectToken } from '../../../generalSlices/userSlice'

interface ReportRowProps {
  report: ReportMetadata
}

const ReportRow = (props: ReportRowProps) => {
  const navigate = useNavigate()
  const token = useSelector(selectToken)
  const { report } = props

  const downloadReport = async (reportName: string) => {
    const promise = ReportsApi.downloadReport({ token: token, reportName })
    toast.promise(promise, {
      loading: `Downloading Performance Report ${reportName}`,
      success: `Successfully Downloaded Performance Report ${reportName}`,
      error: `Error Downloading Performance Report ${reportName}`,
    })
    const report = await promise
    const name = `Performance Report ${reportName}.pdf`
    if (report !== undefined) {
      downloadPdf(report, name)
    }
  }

  const downloadPdf = (contents: Blob, name: string) => {
    const url = window.URL.createObjectURL(contents)
    const link = document.createElement('a')
    link.href = url
    link.setAttribute('download', name) //or any other extension
    document.body.appendChild(link)
    link.click()
  }

  return (
    <TableRow
      key={report.reportName}
      sx={{
        boxShadow: 1,
        transition: (theme) =>
          theme.transitions.create('box-shadow', {
            easing: theme.transitions.easing.easeOut,
          }),
        '&:hover': {
          boxShadow: 8,
        },
        '& > td': {
          backgroundColor: 'background.paper',
          borderBottom: 0,
        },
      }}
    >
      <TableCell width="25%">
        <Box
          component="a"
          sx={{
            alignItems: 'center',
            display: 'inline-flex',
            textDecoration: 'none',
            whiteSpace: 'nowrap',
          }}
          onClick={() => navigate(`/dashboard/logs/1`)}
        >
          <Box sx={{ ml: 2 }}>
            <Typography color="textSecondary" variant="subtitle2">
              {report.reportName}
            </Typography>
          </Box>
        </Box>
      </TableCell>
      <TableCell>
        <Box
          sx={{
            display: 'flex',
            flexDirection: 'column',
          }}
        >
          <Typography variant="subtitle2">Report Duration</Typography>
          <Typography color="textSecondary" variant="body2">
            {report.reportStartTime && format(new Date(report.reportStartTime), 'MMM dd, h:mm:ss a')} -{' '}
            {report.reportStopTime && format(new Date(report.reportStopTime), 'MMM dd, h:mm:ss a')}
          </Typography>
        </Box>
      </TableCell>
      <TableCell>
        <Box
          sx={{
            display: 'flex',
            flexDirection: 'column',
          }}
        >
          <Typography variant="subtitle2">Generated At</Typography>
          <Typography color="textSecondary" variant="body2">
            {report.reportGeneratedAt && format(new Date(report.reportGeneratedAt), 'MMM dd, h:mm:ss a')}
          </Typography>
        </Box>
      </TableCell>
      <TableCell align="right">
        <Button
          endIcon={<ArrowDownward fontSize="small" />}
          onClick={() => {
            downloadReport(report.reportName)
          }}
        >
          Download
        </Button>
      </TableCell>
    </TableRow>
  )
}

interface ReportListTableProps {
  group: boolean
  reports: ReportMetadata[]
  reportsCount: number
  onPageChange: (event: React.MouseEvent<HTMLButtonElement> | null, newPage: number) => void
  onRowsPerPageChange: (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void
  page: number
  rowsPerPage: number
}

export const ReportListTable = (props: ReportListTableProps) => {
  const { group, reports, reportsCount, onPageChange, onRowsPerPageChange, page, rowsPerPage, ...other } = props

  return (
    <div {...other}>
      <PerfectScrollbar>
        <Table
          sx={{
            borderCollapse: 'separate',
            borderSpacing: (theme) => `0 ${theme.spacing(3)}`,
            minWidth: 600,
            marginTop: (theme) => `-${theme.spacing(3)}`,
            p: '1px',
          }}
        >
          {
            <TableBody>
              {reports.map((report: ReportMetadata) => (
                <ReportRow report={report} />
              ))}
            </TableBody>
          }
        </Table>
      </PerfectScrollbar>
      <TablePagination
        component="div"
        count={reportsCount}
        onPageChange={onPageChange}
        onRowsPerPageChange={onRowsPerPageChange}
        page={page}
        rowsPerPage={rowsPerPage}
        rowsPerPageOptions={[5, 10, 25]}
      />
    </div>
  )
}
