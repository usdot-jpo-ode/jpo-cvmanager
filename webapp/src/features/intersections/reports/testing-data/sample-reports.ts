import { ReportMetadata } from '../../../../apis/intersections/reports-api'

export const sampleReports: ReportMetadata[] = [
  {
    reportName: 'Intersection_1234_Weekly_Assessment',
    intersectionID: 1234,
    roadRegulatorID: 'RR-789',
    reportGeneratedAt: new Date('2025-03-15T14:30:00Z'),
    reportStartTime: new Date('2025-03-08T12:00:00Z'),
    reportStopTime: new Date('2025-03-15T12:00:00Z'),
    reportContents: ['Lane Direction of Travel', 'Signal State Events', 'Connection of Travel'],
    laneDirectionOfTravelEventCounts: [
      { id: '1', count: 347 },
      { id: '2', count: 289 },
      { id: '3', count: 412 },
    ],
    laneDirectionOfTravelMedianDistanceDistribution: [
      { id: '0-0.5', count: 145 },
      { id: '0.5-1.0', count: 237 },
      { id: '1.0-1.5', count: 402 },
      { id: '1.5+', count: 264 },
    ],
    laneDirectionOfTravelMedianHeadingDistribution: [
      { id: '0-2', count: 324 },
      { id: '2-5', count: 485 },
      { id: '5-10', count: 198 },
      { id: '10+', count: 41 },
    ],
    laneDirectionOfTravelReportData: [
      { timestamp: 1743434775000, laneID: 1, segmentID: 1, headingDelta: 2.3, medianCenterlineDistance: 0.78 },
      { timestamp: 1743434785000, laneID: 2, segmentID: 2, headingDelta: 1.1, medianCenterlineDistance: 0.45 },
      { timestamp: 1743434795000, laneID: 3, segmentID: 1, headingDelta: 3.6, medianCenterlineDistance: 1.23 },
      { timestamp: 1743434805000, laneID: 1, segmentID: 3, headingDelta: 0.9, medianCenterlineDistance: 0.31 },
    ],
    connectionOfTravelEventCounts: [
      { id: 'valid', count: 2358 },
      { id: 'invalid', count: 167 },
    ],
    signalStateConflictEventCount: [{ id: 'total', count: 24 }],
    signalStateEventCounts: [
      { id: 'red', count: 1024 },
      { id: 'yellow', count: 326 },
      { id: 'green', count: 1452 },
    ],
    signalStateStopEventCounts: [
      { id: 'stopped', count: 957 },
      { id: 'no-stop', count: 214 },
    ],
    timeChangeDetailsEventCount: [{ id: 'total', count: 128 }],
    intersectionReferenceAlignmentEventCounts: [
      { id: 'aligned', count: 6793 },
      { id: 'misaligned', count: 82 },
    ],
    mapBroadcastRateEventCount: [{ id: 'total', count: 84 }],
    mapMinimumDataEventCount: [{ id: 'total', count: 37 }],
    spatMinimumDataEventCount: [{ id: 'total', count: 52 }],
    spatBroadcastRateEventCount: [{ id: 'total', count: 78 }],
    latestMapMinimumDataEventMissingElements: ['laneWidth', 'speedLimit'],
    latestSpatMinimumDataEventMissingElements: ['signalPhaseTime'],
    validConnectionOfTravelData: [
      { connectionID: 1, ingressLaneID: 101, egressLaneID: 201, eventCount: 845 },
      { connectionID: 2, ingressLaneID: 102, egressLaneID: 202, eventCount: 762 },
      { connectionID: 3, ingressLaneID: 103, egressLaneID: 203, eventCount: 751 },
    ],
    invalidConnectionOfTravelData: [
      { connectionID: 4, ingressLaneID: 104, egressLaneID: 204, eventCount: 48 },
      { connectionID: 5, ingressLaneID: 105, egressLaneID: 205, eventCount: 62 },
      { connectionID: 6, ingressLaneID: 106, egressLaneID: 206, eventCount: 57 },
    ],
    headingTolerance: 5,
    distanceTolerance: 1.5,
    stopLineStopReportData: [
      {
        signalGroup: 1,
        numberOfEvents: 328,
        timeStoppedOnRed: 3254,
        timeStoppedOnYellow: 142,
        timeStoppedOnGreen: 65,
        timeStoppedOnDark: 12,
      },
      {
        signalGroup: 2,
        numberOfEvents: 297,
        timeStoppedOnRed: 2987,
        timeStoppedOnYellow: 112,
        timeStoppedOnGreen: 43,
        timeStoppedOnDark: 8,
      },
    ],
    stopLinePassageReportData: [
      {
        signalGroup: 1,
        totalEvents: 643,
        redEvents: 24,
        yellowEvents: 87,
        greenEvents: 532,
        darkEvents: 0,
      },
      {
        signalGroup: 2,
        totalEvents: 586,
        redEvents: 18,
        yellowEvents: 63,
        greenEvents: 505,
        darkEvents: 0,
      },
    ],
  },

  {
    reportName: 'Intersection_5678_Daily_Assessment',
    intersectionID: 5678,
    roadRegulatorID: 'RR-456',
    reportGeneratedAt: new Date('2025-04-01T08:15:00Z'),
    reportStartTime: new Date('2025-03-31T00:00:00Z'),
    reportStopTime: new Date('2025-04-01T00:00:00Z'),
    reportContents: ['Lane Direction of Travel', 'MAP Broadcast', 'SPaT Broadcast'],
    laneDirectionOfTravelEventCounts: [
      { id: '1', count: 124 },
      { id: '2', count: 156 },
    ],
    laneDirectionOfTravelMedianDistanceDistribution: [
      { id: '0-0.5', count: 56 },
      { id: '0.5-1.0', count: 89 },
      { id: '1.0-1.5', count: 78 },
      { id: '1.5+', count: 57 },
    ],
    laneDirectionOfTravelMedianHeadingDistribution: [
      { id: '0-2', count: 112 },
      { id: '2-5', count: 87 },
      { id: '5-10', count: 62 },
      { id: '10+', count: 19 },
    ],
    laneDirectionOfTravelReportData: [
      { timestamp: 1743521175000, laneID: 1, segmentID: 1, headingDelta: 1.7, medianCenterlineDistance: 0.67 },
      { timestamp: 1743521185000, laneID: 2, segmentID: 1, headingDelta: 2.2, medianCenterlineDistance: 0.88 },
      { timestamp: 1743521195000, laneID: 1, segmentID: 2, headingDelta: 1.5, medianCenterlineDistance: 0.54 },
    ],
    connectionOfTravelEventCounts: [
      { id: 'valid', count: 468 },
      { id: 'invalid', count: 32 },
    ],
    signalStateConflictEventCount: [{ id: 'total', count: 4 }],
    signalStateEventCounts: [
      { id: 'red', count: 286 },
      { id: 'yellow', count: 68 },
      { id: 'green', count: 325 },
    ],
    signalStateStopEventCounts: [
      { id: 'stopped', count: 243 },
      { id: 'no-stop', count: 43 },
    ],
    timeChangeDetailsEventCount: [{ id: 'total', count: 32 }],
    intersectionReferenceAlignmentEventCounts: [
      { id: 'aligned', count: 1285 },
      { id: 'misaligned', count: 15 },
    ],
    mapBroadcastRateEventCount: [{ id: 'total', count: 18 }],
    mapMinimumDataEventCount: [{ id: 'total', count: 7 }],
    spatMinimumDataEventCount: [{ id: 'total', count: 12 }],
    spatBroadcastRateEventCount: [{ id: 'total', count: 21 }],
    latestMapMinimumDataEventMissingElements: ['laneOffset'],
    latestSpatMinimumDataEventMissingElements: [],
    validConnectionOfTravelData: [
      { connectionID: 1, ingressLaneID: 101, egressLaneID: 201, eventCount: 248 },
      { connectionID: 2, ingressLaneID: 102, egressLaneID: 202, eventCount: 220 },
    ],
    invalidConnectionOfTravelData: [
      { connectionID: 3, ingressLaneID: 103, egressLaneID: 203, eventCount: 16 },
      { connectionID: 4, ingressLaneID: 104, egressLaneID: 204, eventCount: 16 },
    ],
    headingTolerance: 6,
    distanceTolerance: 2.0,
    stopLineStopReportData: [
      {
        signalGroup: 1,
        numberOfEvents: 142,
        timeStoppedOnRed: 1456,
        timeStoppedOnYellow: 45,
        timeStoppedOnGreen: 12,
        timeStoppedOnDark: 0,
      },
      {
        signalGroup: 2,
        numberOfEvents: 101,
        timeStoppedOnRed: 954,
        timeStoppedOnYellow: 37,
        timeStoppedOnGreen: 6,
        timeStoppedOnDark: 0,
      },
    ],
    stopLinePassageReportData: [
      {
        signalGroup: 1,
        totalEvents: 187,
        redEvents: 6,
        yellowEvents: 23,
        greenEvents: 158,
        darkEvents: 0,
      },
      {
        signalGroup: 2,
        totalEvents: 164,
        redEvents: 4,
        yellowEvents: 17,
        greenEvents: 143,
        darkEvents: 0,
      },
    ],
  },

  {
    reportName: 'Intersection_9012_Monthly_Assessment',
    intersectionID: 9012,
    roadRegulatorID: 'RR-123',
    reportGeneratedAt: new Date('2025-03-31T23:45:00Z'),
    reportStartTime: new Date('2025-03-01T12:00:00Z'),
    reportStopTime: new Date('2025-04-01T12:00:00Z'),
    reportContents: ['All Assessment Types'],
    laneDirectionOfTravelEventCounts: [
      { id: '1', count: 3468 },
      { id: '2', count: 2956 },
      { id: '3', count: 3187 },
      { id: '4', count: 2745 },
    ],
    laneDirectionOfTravelMedianDistanceDistribution: [
      { id: '0-0.5', count: 4235 },
      { id: '0.5-1.0', count: 3876 },
      { id: '1.0-1.5', count: 2654 },
      { id: '1.5+', count: 1591 },
    ],
    laneDirectionOfTravelMedianHeadingDistribution: [
      { id: '0-2', count: 5842 },
      { id: '2-5', count: 3789 },
      { id: '5-10', count: 1986 },
      { id: '10+', count: 739 },
    ],
    laneDirectionOfTravelReportData: [
      { timestamp: 1741743375000, laneID: 1, segmentID: 1, headingDelta: 1.2, medianCenterlineDistance: 0.43 },
      { timestamp: 1741743675000, laneID: 2, segmentID: 1, headingDelta: 2.5, medianCenterlineDistance: 0.78 },
      { timestamp: 1741743975000, laneID: 3, segmentID: 2, headingDelta: 1.8, medianCenterlineDistance: 0.65 },
      { timestamp: 1741744275000, laneID: 4, segmentID: 2, headingDelta: 3.2, medianCenterlineDistance: 0.92 },
    ],
    connectionOfTravelEventCounts: [
      { id: 'valid', count: 29456 },
      { id: 'invalid', count: 1842 },
    ],
    signalStateConflictEventCount: [{ id: 'total', count: 183 }],
    signalStateEventCounts: [
      { id: 'red', count: 10256 },
      { id: 'yellow', count: 3248 },
      { id: 'green', count: 17984 },
    ],
    signalStateStopEventCounts: [
      { id: 'stopped', count: 9654 },
      { id: 'no-stop', count: 602 },
    ],
    timeChangeDetailsEventCount: [{ id: 'total', count: 1284 }],
    intersectionReferenceAlignmentEventCounts: [
      { id: 'aligned', count: 64285 },
      { id: 'misaligned', count: 957 },
    ],
    mapBroadcastRateEventCount: [{ id: 'total', count: 324 }],
    mapMinimumDataEventCount: [{ id: 'total', count: 148 }],
    spatMinimumDataEventCount: [{ id: 'total', count: 186 }],
    spatBroadcastRateEventCount: [{ id: 'total', count: 289 }],
    latestMapMinimumDataEventMissingElements: ['speedLimit', 'maneuverAssist', 'laneWidth'],
    latestSpatMinimumDataEventMissingElements: ['signalGroupID', 'signalPhaseTime'],
    validConnectionOfTravelData: [
      { connectionID: 1, ingressLaneID: 101, egressLaneID: 201, eventCount: 7845 },
      { connectionID: 2, ingressLaneID: 102, egressLaneID: 202, eventCount: 7284 },
      { connectionID: 3, ingressLaneID: 103, egressLaneID: 203, eventCount: 7621 },
      { connectionID: 4, ingressLaneID: 104, egressLaneID: 204, eventCount: 6706 },
    ],
    invalidConnectionOfTravelData: [
      { connectionID: 5, ingressLaneID: 105, egressLaneID: 205, eventCount: 546 },
      { connectionID: 6, ingressLaneID: 106, egressLaneID: 206, eventCount: 487 },
      { connectionID: 7, ingressLaneID: 107, egressLaneID: 207, eventCount: 512 },
      { connectionID: 8, ingressLaneID: 108, egressLaneID: 208, eventCount: 297 },
    ],
    headingTolerance: 4,
    distanceTolerance: 1.2,
    stopLineStopReportData: [
      {
        signalGroup: 1,
        numberOfEvents: 2487,
        timeStoppedOnRed: 32456,
        timeStoppedOnYellow: 1248,
        timeStoppedOnGreen: 372,
        timeStoppedOnDark: 65,
      },
      {
        signalGroup: 2,
        numberOfEvents: 2354,
        timeStoppedOnRed: 29785,
        timeStoppedOnYellow: 1087,
        timeStoppedOnGreen: 289,
        timeStoppedOnDark: 42,
      },
      {
        signalGroup: 3,
        numberOfEvents: 2568,
        timeStoppedOnRed: 33256,
        timeStoppedOnYellow: 1386,
        timeStoppedOnGreen: 298,
        timeStoppedOnDark: 58,
      },
      {
        signalGroup: 4,
        numberOfEvents: 2245,
        timeStoppedOnRed: 27856,
        timeStoppedOnYellow: 984,
        timeStoppedOnGreen: 254,
        timeStoppedOnDark: 37,
      },
    ],
    stopLinePassageReportData: [
      {
        signalGroup: 1,
        totalEvents: 6578,
        redEvents: 234,
        yellowEvents: 865,
        greenEvents: 5479,
        darkEvents: 0,
      },
      {
        signalGroup: 2,
        totalEvents: 5946,
        redEvents: 187,
        yellowEvents: 748,
        greenEvents: 5011,
        darkEvents: 0,
      },
      {
        signalGroup: 3,
        totalEvents: 6243,
        redEvents: 214,
        yellowEvents: 812,
        greenEvents: 5217,
        darkEvents: 0,
      },
      {
        signalGroup: 4,
        totalEvents: 5689,
        redEvents: 165,
        yellowEvents: 698,
        greenEvents: 4826,
        darkEvents: 0,
      },
    ],
  },
]
