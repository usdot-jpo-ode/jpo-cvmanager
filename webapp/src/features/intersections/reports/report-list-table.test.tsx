import React from 'react'
import { render } from '@testing-library/react'
import { ReportListTable } from './report-list-table'
import { Provider } from 'react-redux'
import { ThemeProvider } from '@mui/material'
import { testTheme } from '../../../styles'
import { setupStore } from '../../../store'
import { replaceChaoticIds } from '../../../utils/test-utils'
import { sampleReports } from './testing-data/sample-reports'
import { BrowserRouter } from 'react-router-dom'

// Mock date-fns format to use a specific timezone
jest.mock('date-fns', () => {
  const originalDateFns = jest.requireActual('date-fns')
  const { toZonedTime, fromZonedTime, format } = jest.requireActual('date-fns-tz')

  function formatDateToCustomString(date: Date): string {
    const options: Intl.DateTimeFormatOptions = {
      month: 'short', // Abbreviated month (e.g., "Mar")
      day: '2-digit', // Day of the month (e.g., "08")
      hour: 'numeric', // Hour (e.g., "1", "12")
      minute: '2-digit', // Minutes (e.g., "05")
      second: '2-digit', // Seconds (e.g., "30")
      hour12: true, // Use 12-hour format with AM/PM
    }

    return new Intl.DateTimeFormat('en-US', options).format(date)
  }

  return {
    ...originalDateFns,
    format: (date: Date, formatString) => formatDateToCustomString(date),
  }
})

it('should take a snapshot', () => {
  const { container } = render(
    <ThemeProvider theme={testTheme}>
      <Provider store={setupStore({})}>
        <BrowserRouter>
          <ReportListTable
            reports={sampleReports}
            reportsCount={sampleReports.length}
            onPageChange={() => {}}
            onRowsPerPageChange={() => {}}
            page={0}
            rowsPerPage={10}
            onViewReport={() => {}}
          />
        </BrowserRouter>
      </Provider>
    </ThemeProvider>
  )

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
