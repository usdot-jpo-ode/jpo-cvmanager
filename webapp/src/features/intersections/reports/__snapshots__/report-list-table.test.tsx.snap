// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div>
    <div
      class="scrollbar-container ps"
    >
      <table
        class="MuiTable-root css-mocked-MuiTable-root"
      >
        <tbody
          class="MuiTableBody-root css-mocked-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-mocked-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
              width="25%"
            >
              <a
                class="MuiBox-root css-xpe1lw"
              >
                <div
                  class="MuiBox-root css-d0uhtl"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-mocked-MuiTypography-root"
                  >
                    Intersection_1234_Weekly_Assessment
                  </h6>
                </div>
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
            >
              <div
                class="MuiBox-root css-j7qwjs"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-mocked-MuiTypography-root"
                >
                  Report Duration
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-mocked-MuiTypography-root"
                >
                  Mar 08, 6:00:00 AM
                   
                  -
                   
                  Mar 15, 7:00:00 AM
                </p>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
            >
              <div
                class="MuiBox-root css-j7qwjs"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-mocked-MuiTypography-root"
                >
                  Generated At
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-mocked-MuiTypography-root"
                >
                  Mar 15, 9:30:00 AM
                </p>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-mocked-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                View
              </button>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-mocked-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
              width="25%"
            >
              <a
                class="MuiBox-root css-xpe1lw"
              >
                <div
                  class="MuiBox-root css-d0uhtl"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-mocked-MuiTypography-root"
                  >
                    Intersection_5678_Daily_Assessment
                  </h6>
                </div>
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
            >
              <div
                class="MuiBox-root css-j7qwjs"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-mocked-MuiTypography-root"
                >
                  Report Duration
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-mocked-MuiTypography-root"
                >
                  Mar 30, 7:00:00 PM
                   
                  -
                   
                  Mar 31, 7:00:00 PM
                </p>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
            >
              <div
                class="MuiBox-root css-j7qwjs"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-mocked-MuiTypography-root"
                >
                  Generated At
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-mocked-MuiTypography-root"
                >
                  Apr 01, 3:15:00 AM
                </p>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-mocked-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                View
              </button>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-mocked-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
              width="25%"
            >
              <a
                class="MuiBox-root css-xpe1lw"
              >
                <div
                  class="MuiBox-root css-d0uhtl"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-mocked-MuiTypography-root"
                  >
                    Intersection_9012_Monthly_Assessment
                  </h6>
                </div>
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
            >
              <div
                class="MuiBox-root css-j7qwjs"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-mocked-MuiTypography-root"
                >
                  Report Duration
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-mocked-MuiTypography-root"
                >
                  Mar 01, 6:00:00 AM
                   
                  -
                   
                  Apr 01, 7:00:00 AM
                </p>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
            >
              <div
                class="MuiBox-root css-j7qwjs"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-mocked-MuiTypography-root"
                >
                  Generated At
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-mocked-MuiTypography-root"
                >
                  Mar 31, 6:45:00 PM
                </p>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-mocked-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                View
              </button>
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="ps__rail-x"
        style="left: 0px; top: 0px;"
      >
        <div
          class="ps__thumb-x"
          style="left: 0px; width: 0px;"
          tabindex="0"
        />
      </div>
      <div
        class="ps__rail-y"
        style="top: 0px; left: 0px;"
      >
        <div
          class="ps__thumb-y"
          style="top: 0px; height: 0px;"
          tabindex="0"
        />
      </div>
    </div>
    <div
      class="MuiTablePagination-root css-mocked-MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-mocked-MuiToolbar-root-MuiTablePagination-toolbar"
      >
        <div
          class="MuiTablePagination-spacer css-mocked-MuiTablePagination-spacer"
        />
        <p
          class="MuiTablePagination-selectLabel css-mocked-MuiTablePagination-selectLabel"
          id="mui-3"
        >
          Rows per page:
        </p>
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-mocked-MuiInputBase-root-MuiTablePagination-select"
        >
          <div
            aria-controls="mui-1"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="mui-3 mui-2"
            class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-mocked-MuiSelect-select-MuiInputBase-input"
            id="mui-2"
            role="combobox"
            tabindex="0"
          >
            10
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-mocked-MuiSelect-nativeInput"
            tabindex="-1"
            value="10"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-mocked-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTablePagination-displayedRows css-mocked-MuiTablePagination-displayedRows"
        >
          1–3 of 3
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mocked-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mocked-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mocked-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mocked-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
